### PR TITLE
Shotgun Ammo Display Tweak

### PR DESCRIPTION
**Pull request by Geevies**
* Shotguns now more accurately reflect shells in the tube and whether there's a shell in the chamber.

image
image

### CODE REVIEW

**Comment by Geevies** [comment]
i should note this is when the holographic ammo display is attached, they don't spawn like this

**Comment by Geevies** [comment]
!review

**Review by SleepyGemmy — APPROVED** [review]

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -1032,15 +1032,19 @@
...
-		var/ammo = get_ammo()
-		if(ammo > 9)
-			if(ammo < 20)
-				maptext_x = 20
-			else
-				maptext_x = 18
+		handle_maptext()
+
+
+/obj/item/gun/proc/handle_maptext()
```
Comment (FluffyGhoster) [inline code/modules/projectiles/gun.dm:1038]: DMDoc this, and mark it as not sleepable

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -67,6 +67,15 @@
 	///Whether the item icon has a cycling animation
 	var/cycle_anim = TRUE
 
+/obj/item/gun/projectile/shotgun/pump/handle_maptext()
+	var/ammo = length(loaded) || 0
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/shotgun.dm:71]: The 0 should not be necessary, this isn't a lazy list and doesn't seem like it's nulled anywhere except deletion (at which point it's not handlign the maptext anymore)

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**